Kernel/Bus/SerialIO/PS2/Controller.h

```diff
@@ -26,6 +26,14 @@ enum class PS2DeviceCommand {
 };
 
 class PS2Controller : public SerialIOController {
+public:
+    virtual ErrorOr<void> reset_device(PS2PortIndex) = 0;
+    virtual ErrorOr<void> send_command(PS2PortIndex, PS2DeviceCommand command) = 0;
+    virtual ErrorOr<void> send_command(PS2PortIndex, PS2DeviceCommand command, u8 data) = 0;
+
+    virtual ErrorOr<u8> read_from_device(PS2PortIndex) = 0;
+    virtual ErrorOr<void> prepare_for_input(PS2PortIndex) = 0;
+    virtual bool irq_process_input_buffer(PS2PortIndex) = 0;
 
```

Comment:
I'm skeptical about the utility of this interface. Is there anyone in the system that can use any arbitrary PS2 device without caring if it's a mouse or a keyboard or something else entirely? That is to say, can I swap out your I8042Device in ... VMWareBackdoor, and it will work the same? Is this interface actually properly *substitutable*? (https://en.wikipedia.org/wiki/Liskov_substitution_principle)

I wonder if the I8042 devices should just be SerialIOControllers directly without this interface class in the middle.